sockeye/inference.py

```diff
@@ -1259,8 +1311,10 @@ def _beam_search(self,
         # (0) encode source sentence, returns a list
         model_states = self._encode(source, source_length)
 
+        # Records items in the beam that are inactive. At the beginning (t==1), there is only one valid or active
+        # item on the beam for each sentence
+        inactive = mx.nd.array(([0] + [1] * (self.beam_size - 1)) * self.batch_size, ctx=self.context, dtype='int32')
```

Comment:
I wonder if 
```
inactive = mx.nd.ones((self.batch_size * self.beam_size), dtype='int32', ctx=self.context)
inactive[::beam_size] = 0
```
wouldn't be easier. It would also reuse the common pattern of shape batch * beam, instead of beam * batch.
And you would avoid the python lists...